x-pack/plugin/sql/qa/jdbc/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/JdbcTestUtils.java

```diff
@@ -57,4 +84,26 @@ static long convertFromCalendarToUTC(long value, Calendar cal) {
 
         return convertedDateTime.toInstant().toEpochMilli();
     }
+
+    static String asStringTimestampFromNanos(long nanos) {
+        return asStringTimestampFromNanos(nanos, ZoneId.systemDefault());
+    }
+
+    static String asStringTimestampFromNanos(long nanos, ZoneId zoneId) {
+        ZonedDateTime zdt = ZonedDateTime.ofInstant(Instant.ofEpochMilli(toMilliSeconds(nanos)), zoneId);
+        return StringUtils.toString(zdt.withNano((int) (nanos % 1_000_000_000)));
+    }
+
+    static long randomNanos() {
+        // Return a number which is at least 20:00:00.000000000 to avoid switching to negative values when a UTC-XX hours is applied
+        return randomLongBetween(72000000000000L, Long.MAX_VALUE);
+    }
+
+    static int extractNanosOnly(long nanos) {
+        return (int) (nanos % 1_000_000_000);
+    }
+
+    static boolean versionSupportsDateNanos() {
+        return JDBC_DRIVER_VERSION.onOrAfter(Version.V_8_0_0);
```

Comment:
I guess you'll need to change this twice, eventually, no?
- in branch, with branch' version (like 7.12);
- in master, once ported in branch (ideally not long after, not to forget), so that 7.13 won't have the 8.0.0 value.

Reply:
yep, thx for the reminder!